### ISSUE TITLE
TypeScript throws an error when trying to use custom compare

### ISSUE THREAD

**Issue by LyonJenkins** [Teun/thenBy.js#37]
When using the custom compare demo code given, TypeScript throws an error.
``` js
const cards = ['2', '3', '4', '5', '6', '7', '8', '9', '10', 'J', 'Q', 'K', 'A'];
var cardCompare = (c1, c2) =>{
    return cards.indexOf(c1) - cards.indexOf(c2);
}
var handOfCards =  [
        { id: 7, suit:"c", card:"A" },
        { id: 8, suit:"d", card:"10" },
        // etc
    ];
handOfCards.sort(firstBy("card", {cmp: cardCompare, direction: "desc"}));
```
```
ERROR in src/app/opcenter/personnelfiles/personnelfiles.component.ts(103,27): error TS2345: Argument of type '{ cmp: (c1: any,
c2: any) => number; }' is not assignable to parameter of type '1 | "desc" | -1 | opt | "asc"'.
      Object literal may only specify known properties, and 'cmp' does not exist in type 'opt'.
```
I have literally done nothing other than copy the demo code. There seems to be an issue going on here.

**Comment by Teun** [comment]
Thank you for signalling that. It seems that this feature (the custom compare function) was completely missed when we created the typescript bindings. The TypeScript compiler indicates that it doesn't expect the `cmp` property.

Will fix this with some priority.

**Comment by Teun** [comment]
I have just published version 1.3.3, which should fix this problem. Could you please test that it does?
